### ISSUE TITLE
Review rule definitions

### ISSUE THREAD

**Issue by ChristopherKiss** [quick-foot/eslint-config#1]
The following rules are not currently playing nicely.

```json
"@typescript-eslint/explicit-member-accessibility": "off"
```
This rule will need https://github.com/typescript-eslint/typescript-eslint/issues/214 to be fixed.

```json
"@typescript-eslint/camelcase": "off"
```
This rule currently doesn't play nice with Github api access_token properties. Need to review if this is something I want.

```
tslint-consistent-codestyle
```
Would like to have support for this for Angular specific rules.

```
return-early
```
Look into a rule that can satisfy this one.

**Comment by gavinbarron** [comment]
I'm working on #214 at present. So I'll see if there's an obvious resolution to the `off` option not working

**Comment by ChristopherKiss** [comment]
@gavinbarron,

Sorry, it's not quite as clear. The off option works fine, that's what I've currently set it to as I can't use this rule. The main issue is that I don't like prefixing constructor with member accessibility, but I like to prefix the rest (properties, methods, accessors), currently eslint doesn't provide the option to exclude the constructor from failing the lint rule as noted in https://github.com/typescript-eslint/typescript-eslint/issues/214.

This is also the difference between the eslint rule and the tslint rule. Eslint does not provide any options, it's all or nothing.

Looking forward to seeing your changes to make this rule that little bit better. Much appreciated.

**Comment by gavinbarron** [comment]
Ahhh, gotcha. I just started writing the tests to allow for no public modifier on a constructor this morning. Will be implementing the code to meet your ask tonight.
